frontend/src/pages/modelRegistry/screens/RegisterModel/RegisterModel.tsx

```diff
@@ -36,12 +37,21 @@ const RegisterModel: React.FC = () => {
   const [isSubmitting, setIsSubmitting] = React.useState(false);
   const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
   const [formData, setData] = useRegisterModelData();
-  const isModelNameValid = isNameValid(formData.modelName);
-  const isSubmitDisabled = isSubmitting || isRegisterModelSubmitDisabled(formData);
-  const { modelName, modelDescription } = formData;
   const [registeredModelName, setRegisteredModelName] = React.useState<string>('');
   const [versionName, setVersionName] = React.useState<string>('');
   const [errorName, setErrorName] = React.useState<string | undefined>(undefined);
+  const [registeredModels] = useRegisteredModels();
```

Comment:
I don't love this because technically we won't correctly validate the name until this fetch resolves, so to be fully correct we'd need to pull out the `isLoaded` from this fetch result and also block submit if it's not loaded yet. But this is probably fine because even if the user submits the form before this resolves (i.e. they type faster than their network, or there's an error fetching the registered models), they'll get the existing error message back, so this is still an improvement and it's very unlikely to be an issue.

Just thinking out loud.... I think it's fine to leave here, but we should be careful fetching without looking at the loading state in general.

Reply:
we'd want to do more than block until loaded, as (if it were ever not loaded by the time the user went to submit) they would not know why we are blocking the submit. there would have to be some sort of indicator, and i'm not sure what that would look like.

Reply:
that's true. not worth implementing that indicator if they're never going to see it. We actually might be better served by showing a loading spinner on the page until the models are loaded - in fact that's easy here, we can just take `loaded` and `loadError` out of this hook call and pass them to `ApplicationsPage`, it already abstracts that rendering for us. That's probably what we should do.